### PR TITLE
Updating Order Cancel condition to unhold order if needed CH-31031

### PR DESCRIPTION
**Pull request by CTDoriott**

### CODE REVIEW

File: Cron/Order.php [hunk 1]
```diff
@@ -190,10 +190,21 @@ protected function processOrderStatusUpdate(OrderInterface $order, string $newSt
     protected function cancelOrder(OrderInterface $order) : bool
     {
         try {
+            $isUnholded = false;
+            if ($order->canUnhold()) {
```
Comment (crfroehlich) [inline Cron/Order.php:194]: I assume this returns true if the order is not holded in the first place?

File: Cron/Order.php [hunk 2]
```diff
@@ -190,10 +190,21 @@ protected function processOrderStatusUpdate(OrderInterface $order, string $newSt
...
+                $order->unhold();
+            }
+
             if (!$order->canCancel()) {
                 $this->loggingClient->info(
                     sprintf('Unable to cancel Order #%s as it cannot be canceled', $order->getIncrementId())
                 );
+
+                if ($isUnholded) {
+                    $order->hold();
```
Comment (crfroehlich) [inline Cron/Order.php:205]: If we can't cancel, can we hold? Are we sure this is the desired behavior?

File: Cron/Order.php (hunk 1 above)
Comment (CTDoriott) [reply Cron/Order.php:194]: @crfroehlich If this returns true then the order is holded in the first place therefore it must be unholded

Comment (davehensleyns8) [reply Cron/Order.php:194]: `canUnhold()` _only_ returns `true` [if the order is holded](https://github.com/magento/magento2/blob/2.3/app/code/Magento/Sales/Model/Order.php#L763).

File: Cron/Order.php (hunk 2 above)
Comment (CTDoriott) [reply Cron/Order.php:205]: @crfroehlich If we can't cancel and the order was originally holded then we need to revert it to the original state of the order so we don't break the Merchant's intended state - This is the desired behavior

File: Cron/Order.php (hunk 1 above)
Comment (crfroehlich) [reply Cron/Order.php:194]: so don't we always want to cancel if the order can be cancelled?

Comment (CTDoriott) [reply Cron/Order.php:194]: We do want to cancel it if it can be canceled thus we need to unhold it then cancel it @crfroehlich

Comment (crfroehlich) [reply Cron/Order.php:194]: Nevermind. Operator error.

Comment (CTDoriott) [reply Cron/Order.php:194]: Coolness, merging

**Pull request merged**